flash/tabular/regression/data.py

```diff
@@ -11,8 +11,165 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from flash.tabular.data import TabularData
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from flash.core.data.callback import BaseDataFetcher
+from flash.core.data.process import Deserializer
+from flash.core.utilities.imports import _PANDAS_AVAILABLE
+
+if _PANDAS_AVAILABLE:
+    from pandas.core.frame import DataFrame
+else:
+    DataFrame = object
+
+from pytorch_forecasting import TimeSeriesDataSet
+
+from flash.tabular.data import TabularData, TabularDataFrameDataSource, TabularPreprocess
+
+
+class TabularRegressionDataFrameDataSource(TabularDataFrameDataSource):
+
+    def __init__(
+        self,
+        time_idx: str,
+        target: Union[str, List[str]],
+        group_ids: List[str],
+        cat_cols: Optional[List[str]] = None,
+        num_cols: Optional[List[str]] = None,
+        target_col: Optional[str] = None,
+        mean: Optional[DataFrame] = None,
+        std: Optional[DataFrame] = None,
+        codes: Optional[Dict[str, Any]] = None,
+        target_codes: Optional[Dict[str, Any]] = None,
+        classes: Optional[List[str]] = None,
+        **data_source_kwargs: Any
+    ):
+        self.time_idx = time_idx
+        self.target = target
+        self.group_ids = group_ids
+        self.data_source_kwargs = data_source_kwargs
+        super(TabularRegressionDataFrameDataSource, self).__init__(
+            cat_cols=cat_cols,
+            num_cols=num_cols,
+            target_col=target_col,
+            mean=mean,
+            std=std,
+            codes=codes,
+            target_codes=target_codes,
+            classes=classes,
+            is_regression=True
+        )
+
+    def load_data(self, data: DataFrame, dataset: Optional[Any] = None):
+        return TimeSeriesDataSet(
+            data, time_idx=self.time_idx, group_ids=self.group_ids, target=self.target, **self.data_source_kwargs
+        )
+
+
+class TabularRegressionPreprocess(TabularPreprocess):

```

Comment:
Same here, maybe should just be a Preprocess? That way you can remove the arguments to __init__ that are not used for regression